### PR TITLE
lib: enable support for zoslib on z/OS

### PR DESCRIPTION
**Pull request by alexcfyung**
Enable support for zoslib, a dependency for node.js on z/OS.

##### Checklist

- [x] `npm install && npm test` passes
- [ ] tests are included 
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

**Review by richardlau — APPROVED** [review]

File: lib/configure.js [hunk 1]
```diff
@@ -280,6 +280,43 @@ function configure (gyp, argv, callback) {
       }
     }
 
+    // For z/OS we need to set up the path to zoslib include directory,
+    // which contains headers included in v8config.h.
+    var zoslibIncDir
+    if (process.platform === 'os390') {
+      logprefix = "find zoslib's zos-base.h:"
+      let msg
```
Comment (zsw007) [inline lib/configure.js:288]: Is it ok for `let` to be used here? It seems like the convention for this file is entirely `var` and `const`.

File: lib/configure.js [hunk 2]
```diff
@@ -280,6 +280,43 @@ function configure (gyp, argv, callback) {
...
+      var zoslibIncTmp = process.env.ZOSLIB_INCLUDES
```
Comment (zsw007) [inline lib/configure.js]: The variable name `zoslibIncTmp` makes me think that the path is in tmpdir. Maybe a better name could be `zoslibEnvar`, `zoslibPath`, or something else?

File: lib/configure.js (hunk 1 above)
Comment (alexcfyung) [reply lib/configure.js:288]: using var will cause linter to complain as msg was defined in line 277

File: lib/configure.js [hunk 3]
```diff
@@ -280,6 +280,43 @@ function configure (gyp, argv, callback) {
...
+          msg = msgFormat('Could not find any of %s in directory %s; set ' +
+                          'environmant variable ZOSLIB_INCLUDES to the path ' +
+                          'that contains zos-base.h', candidates.toString(), nodeRootDir)
+        }
+      }
+      if (zoslibIncPath !== undefined) {
+        zoslibIncDir = path.dirname(zoslibIncPath)
+        log.verbose(logprefix, "Found zoslib's zos-base.h in: %s", zoslibIncDir)
+      } else {
+        log.error(logprefix, msg)
```
Comment (IgorTodorovskiIBM) [inline lib/configure.js:316]: Can we make this apply to Node v16 and above since we don't ship zoslib in earlier versions?

Comment (alexcfyung) [reply lib/configure.js:316]: would it be sufficient to state in the description that only merge this change to v16 or above instead of manual checking?

Comment (richardlau) [reply lib/configure.js:316]: No because node-gyp should work on all supported versions of Node.js.

File: lib/configure.js [hunk 4]
```diff
@@ -311,7 +311,8 @@ function configure (gyp, argv, callback) {
       if (zoslibIncPath !== undefined) {
         zoslibIncDir = path.dirname(zoslibIncPath)
         log.verbose(logprefix, "Found zoslib's zos-base.h in: %s", zoslibIncDir)
-      } else {
+      } else if (release.version.split('.')[0] >= 16) {
+        // zoslib is only shipped after Node v16
```
Comment (gabylb) [inline lib/configure.js]:
```suggestion
        // zoslib is only shipped in Node v16 and above.
```

**Review by gabylb — APPROVED** [review]
LGTM

**Review by zsw007 — APPROVED** [review]
LGTM

File: lib/configure.js [hunk 5]
```diff
@@ -307,6 +345,9 @@ function configure (gyp, argv, callback) {
       argv.push('-Dnode_root_dir=' + nodeDir)
       if (process.platform === 'aix' || process.platform === 'os390') {
         argv.push('-Dnode_exp_file=' + nodeExpFile)
+        if (process.platform === 'os390') {
+          argv.push('-Dzoslib_include_dir=' + zoslibIncDir)
```
Comment (IgorTodorovskiIBM) [inline lib/configure.js:349]: Will zoslibIncDir be undefined here for Node release < 16.0?

**Review by IgorTodorovskiIBM — APPROVED** [review]
LGTM

**Pull request merged**